vnext/JSI/Shared/ChakraRuntime.cpp

```diff
@@ -226,211 +270,195 @@ facebook::jsi::Runtime::PointerValue *ChakraRuntime::clonePropNameID(
 }
 
 facebook::jsi::PropNameID ChakraRuntime::createPropNameIDFromAscii(const char *str, size_t length) {
-  ChakraObjectRef propId = GetPropertyId(std::string_view{str, length});
-  return MakePointer<facebook::jsi::PropNameID>(std::move(propId));
+  JsPropertyIdRef propertyId = GetPropertyIdFromName(std::string_view{str, length});
+  return MakePointer<facebook::jsi::PropNameID>(propertyId);
 }
 
 facebook::jsi::PropNameID ChakraRuntime::createPropNameIDFromUtf8(const uint8_t *utf8, size_t length) {
-  return createPropNameIDFromAscii(Common::Utilities::CheckedReinterpretCast<const char *>(utf8), length);
+  JsPropertyIdRef propertyId = GetPropertyIdFromName(std::string_view{reinterpret_cast<const char *>(utf8), length});
+  return MakePointer<facebook::jsi::PropNameID>(propertyId);
 }
 
 facebook::jsi::PropNameID ChakraRuntime::createPropNameIDFromString(const facebook::jsi::String &str) {
-  // We don not use the functions:
-  //   std::string ChakraRuntime::utf8(const String& str), and
-  //   std::string ToStdString(const ChakraObjectRef &jsString)
-  // here to avoud excessive Unicode conversions.
-  std::wstring propName = ToStdWstring(GetChakraObjectRef(str));
-  return MakePointer<facebook::jsi::PropNameID>(GetPropertyId(propName));
+  JsPropertyIdRef propertyId = GetPropertyIdFromName(StringToPointer(GetJsRef(str)).data());
+  return MakePointer<facebook::jsi::PropNameID>(propertyId);
 }
 
 std::string ChakraRuntime::utf8(const facebook::jsi::PropNameID &id) {
-  return Common::Unicode::Utf16ToUtf8(GetPropertyName(GetChakraObjectRef(id)));
+  return Common::Unicode::Utf16ToUtf8(GetPropertyNameFromId(GetJsRef(id)));
 }
 
 bool ChakraRuntime::compare(const facebook::jsi::PropNameID &lhs, const facebook::jsi::PropNameID &rhs) {
-  return CompareJsPropertyIds(GetChakraObjectRef(lhs), GetChakraObjectRef(rhs));
+  return GetJsRef(lhs) == GetJsRef(rhs);
 }
 
 std::string ChakraRuntime::symbolToString(const facebook::jsi::Symbol &s) {
-  return ToStdString(ToJsString(GetChakraObjectRef(s)));
+  JsValueRef symbol = GetJsRef(s);
+  JsValueRef symbolCtor = GetProperty(GetGlobalObject(), m_propertyId.Symbol);
+  JsValueRef symbolPrototype = GetProperty(symbolCtor, m_propertyId.prototype);
+  JsValueRef symbolToString = GetProperty(symbolPrototype, m_propertyId.toString);
+  JsValueRef jsString = CallFunction(symbolToString, {symbol});
+  return StringToStdString(jsString);
```

Comment:
these can be auto (since you don't care about their types?) and const

Reply:
Using a concrete type is always preferable for better readability. I will check the use of const. The JsValueRef is an alias for void*. Use of const may cause it become 'const void*' that it makes it unacceptable as a parameter.

---
In reply to: [470467566](https://github.com/microsoft/react-native-windows/pull/5710#discussion_r470467566) [](ancestors = 470467566)

Reply:
There was no issue using const :)

---
In reply to: [470700938](https://github.com/microsoft/react-native-windows/pull/5710#discussion_r470700938) [](ancestors = 470700938,470467566)